cli/monitor/monitor.go

```diff
@@ -0,0 +1,107 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2020 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package monitor
+
+import (
+	"context"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/arduino/arduino-cli/cli/arguments"
+	"github.com/arduino/arduino-cli/cli/errorcodes"
+	"github.com/arduino/arduino-cli/cli/feedback"
+	"github.com/arduino/arduino-cli/cli/instance"
+	"github.com/arduino/arduino-cli/commands/monitor"
+	"github.com/arduino/arduino-cli/i18n"
+	rpc "github.com/arduino/arduino-cli/rpc/cc/arduino/cli/commands/v1"
+	"github.com/arduino/arduino-cli/table"
+	"github.com/fatih/color"
+	"github.com/spf13/cobra"
+)
+
+var tr = i18n.Tr
+
+var portArgs arguments.Port
+var describe bool
+
+// NewCommand created a new `monitor` command
+func NewCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "monitor",
+		Short: tr("Open a communication port with a board."),
+		Long:  tr("Open a communication port with a board."),
+		Example: "" +
+			"  " + os.Args[0] + " monitor -p /dev/ttyACM0\n" +
+			"  " + os.Args[0] + " monitor -p /dev/ttyACM0 --describe",
+		Run: runMonitorCmd,
+	}
+	portArgs.AddToCommand(cmd)
+	cmd.Flags().BoolVar(&describe, "describe", false, tr("Show all the settings of the communication port."))
+	cmd.MarkFlagRequired("port")
+	return cmd
+}
+
+func runMonitorCmd(cmd *cobra.Command, args []string) {
+	instance := instance.CreateAndInit()
+
+	port, err := portArgs.GetPort(instance, nil)
+	if err != nil {
+		if err != nil {
+			feedback.Error(err)
+			os.Exit(errorcodes.ErrGeneric)
+		}
+	}
```

Comment:
We wanted to be sure there's an error for real I guess. :sweat_smile:

Reply:
Better checking two times than one!
LOL